src/main/java/org/ow2/proactive/workflow_catalog/rest/service/WorkflowRevisionService.java

```diff
@@ -126,11 +126,13 @@ public WorkflowMetadata createWorkflowRevision(Long bucketId, Optional<Long> wor
     @Transactional
     public WorkflowMetadata createWorkflowRevision(Long bucketId, Optional<Long> workflowId,
             ProActiveWorkflowParserResult proActiveWorkflowParserResult,
+            Optional<String> layout,
             byte[] proActiveWorkflowXmlContent) {
         Bucket bucket = findBucket(bucketId);
 
         Workflow workflow = null;
         WorkflowRevision workflowRevision;
+        String layoutValue = layout.isPresent() ? layout.get() : "";
```

Comment:
You can use the orElse functionality from the Optional monad here : 

String layoutValue = layout.orElse("")